### PR TITLE
Decouple fetching layer from Redux, part 2

### PR DESCRIPTION
**Pull request by axelboc**
I'm extracting a few fetch calls and cleaning up an unused action related to the Beamline API .

I'm also settling down on a naming convention and applying it to the Login API fetching layer introduced in #1124:
  - `fetch<Resource>` for `GET` requests that retrieve resources (e.g. `fetchLoginInfo`, `fetchBeamlineSetup`) — and `get<Resource>` for their corresponding Redux action creators,
  - `send<PerformAction>` for API endpoints that perform actions on the server regardless of the HTTP method used (e.g. `sendLogIn`, `sendRefreshSession`, `sendPrepareBeamlineForNewSample`) — and simply `<performAction>` for their corresponding Redux action creators.

### CODE REVIEW

File: mxcubeweb/routes/beamline.py [hunk 1]
```diff
@@ -182,13 +182,13 @@ def beamline_abort_action(name):
 
         :param str name: Owner / Actuator of the process/action to abort
 
-        Replies with status code 200 on success and 520 on exceptions.
```
Comment (axelboc) [inline mxcubeweb/routes/beamline.py:185]: 520 does not seem to be an official HTTP error code (it's used by Cloudflare, apparently). I think 500 is more appropriate?

File: ui/src/actions/beamline.js [hunk 2]
```diff
@@ -46,30 +47,6 @@ export function updateBeamlineHardwareObjectStateAction(data) {
   return { type: BL_UPDATE_HARDWARE_OBJECT_STATE, data };
 }
 
-export function sendGetAllhardwareObjects() {
```
Comment (axelboc) [inline ui/src/actions/beamline.js:49]: From what I could tell, this action was not actually dispatched. The fetch call is duplicated and invoked directly in `general.js`.

File: ui/src/containers/BeamlineSetupContainer.jsx [hunk 3]
```diff
@@ -323,10 +319,6 @@ function mapStateToProps(state) {
 
 function mapDispatchToProps(dispatch) {
   return {
-    getAllhardwareObjects: bindActionCreators(
```
Comment (axelboc) [inline ui/src/containers/BeamlineSetupContainer.jsx:326]: The unused action I mentioned was bound in this container but never dispatched from any of the child components (unless there's a weird prop access happening that I've missed, like `this.props['getAll' + 'hardwareObjects']`... :joy:).

Comment (marcus-oscarsson) [reply ui/src/containers/BeamlineSetupContainer.jsx:326]: I don't think we are accessing any prop like that, other-tings perhaps but not props :)

File: mxcubeweb/routes/beamline.py (hunk 1 above)
Comment (marcus-oscarsson) [reply mxcubeweb/routes/beamline.py:185]: Agreed

**Comment by marcus-oscarsson** [comment]
Thumbs up for the naming convention, seems very reasonable to me. I guess we can make a separate PR for the error return code and the redirect that you spotted in the previous PR.

**Comment by marcus-oscarsson** [comment]
Issue here: https://github.com/mxcube/mxcubeweb/issues/1126

**Comment by marcus-oscarsson** [comment]
Looks good to me !

**Review by marcus-oscarsson — APPROVED** [review]

**Pull request merged**